### PR TITLE
Added @example tags to src/instance.js

### LINKED ISSUE #157: @example Comments should refer to sample code
nodejs-bigtable has `@example` comments that include code.  This is error prone, since it's not compiled and may get obsolete.  Instead of inline code, all `@example`s should refer to a sample file and a tag within it.

nodejs-spanner has a great examples of this.  Here's one example ([link](https://github.com/googleapis/nodejs-spanner/blob/72efac10bcab961732cca234312908fa1b5bc3dd/src/batch-transaction.js#L116)):

```
/**
 ...
 * @example <caption>include:samples/batch.js</caption>
 * region_tag:spanner_batch_client
*/
```

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes #157 for Instance.js (it's a good idea to open an issue first for discussion)

- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by sduskis** [comment]
@jabubake: Do we need tests for these snippets?

**Comment by jabubake** [comment]
@sduskis : Yes, ensuring these are tested will help maintain them.

Since these snippets are for the API methods, one option to minimize the number of tests is to ensure that the [integration tests](https://github.com/googleapis/nodejs-bigtable/blob/master/system-test/bigtable.js) for the library leverage these snippets as test cases.

**Comment by stephenplusplus** [comment]
Samples actually have their own testing files and conventions: https://github.com/googleapis/nodejs-bigtable/tree/master/samples. See the `system-test` directory.

There is already a file, `instances.js`. Maybe these new tests should be moved into that file, then tested from `system-test/instances.test.js`?

**Comment by jabubake** [comment]
@stephenplusplus I agree there is overlap here : in this case, `samples/instance.js` provides a clean way to run through basic use cases in a single format. The snippets are intended to focus on the the different ways of  calling the API w/ promise, callback etc. This is a little unique to node.js, so i'm fine with the additional source file here vs cluttering `samples/instance.js` : we can maybe remove the cli from the `document-snippets/instance.js` and just have tests written against it, so that developers do not run into weird states running the CLI  commands out of order.

File: samples/document-snippets/instance.js [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const Bigtable = require('@google-cloud/bigtable');
+const bigtableClient = new Bigtable();
+
+function newInstance(instanceId) {
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Please call these functions something similar to the tag.  In this case, please use `createInstance`.

File: samples/document-snippets/instance.js [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+  // creates a new Instance
+  instance
+    .create(options)
+    .then(result => {
+      let newInstance = result[0];
+      // let operations = result[1];
+      // let apiResponse = result[2];
+
+      console.log(`Created Instance: ${newInstance.name}`);
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Please use `id` here.

File: samples/document-snippets/instance.js [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+    storage: 'ssd',
+  };
+
+  instance
+    .createCluster(clusterId, options)
+    .then(result => {
+      const newCluster = result[0];
+      // const operations = result[1];
+      // const apiResponse = result[2];
+      console.log(`Cluster created: ${newCluster.name}`);
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Please use `id` here.

File: samples/document-snippets/instance.js [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+
+      console.log(`Created Instance: ${newInstance.name}`);
+    })
+    .catch(err => {
+      console.error('Error creating prod-instance:', err);
+    });
+  // [END bigtable_create_instance]
+}
+
+function newCluster(clusterId, instanceId) {
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Please call this `createCluster`.

File: samples/document-snippets/instance.js [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+      // const apiResponse = result[2];
+      console.log(`Cluster created: ${newCluster.name}`);
+    })
+    .catch(err => {
+      console.error('Error creating cluster: ', err);
+    });
+  // [END bigtable_create_cluster]
+}
+
+function newAppProfile(appProfileId, clusterId, instanceId) {
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Please call this `createAppProfile`.  Also, please reverse the order of parameters.  This should be `instanceId, clusterId, appProfileId`.

File: samples/document-snippets/instance.js (hunk 4 above)
Comment (sduskis) [reply samples/document-snippets/instance.js]: Also, please start the parameters from the "greatest" thing.  In this case, please use `instanceId, clusterId`

File: samples/document-snippets/instance.js [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+    .catch(err => {
+      console.error('Error geting AppProfiles: ', err);
+    });
+  // [END bigtable_get_app_profiles]
+}
+
+function getMetaData(instanceId) {
+  const instance = bigtableClient.instance(instanceId);
+
+  // [START bigtable_get_imeta]
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Please use the full name in the tag, specifically, `bigtable_get_instance_metadata`.

File: samples/document-snippets/instance.js [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+  //   autoPaginate: true
+  // };
+
+  instance
+    .getTables(options)
+    .then(result => {
+      console.log(`Tables:`);
+      let tables = result[0];
+      tables.forEach(t => {
+        console.log(t.name);
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Please use `t.id`

File: samples/document-snippets/instance.js [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+      });
+    })
+    .catch(err => {
+      console.error('Error geting Tables: ', err);
+    });
+  // [END bigtable_get_tables]
+}
+
+function setMetaData(instanceId) {
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Please call this method `updateInstance`

Comment (vijay-qlogic) [reply samples/document-snippets/instance.js]: Done
https://github.com/googleapis/nodejs-bigtable/pull/222/commits/174f67cadb820ec8b26349131f69e66a403fc720

**Comment by sduskis** [comment]
@stephenplusplus, do you know how to generate API documentation?  That's important in testing this PR.

**Comment by stephenplusplus** [comment]
That's all done behind the scenes from me. @jmdobry can probably help.

**Comment by vijay-qlogic** [comment]
I searched for generating API docs that pull in examples from other files, but no where in **JSDoc** guide found `region_tag` being used to for same.
@jmdobry can you please help here.

**Comment by jmdobry** [comment]
If your changes were on a branch of the nodejs-bigtable repo itself, I could stage the API ref docs to test their use of `region_tag`. I don't have a way to test when the changes are on a fork.

**Comment by sduskis** [comment]
@jmdobry, can you potentially use  `vijay-qlogic:tag-example-comments` as the branch?

**Comment by vijay-qlogic** [comment]
@jmdobry I have referred [nodejs-spanner](https://github.com/googleapis/nodejs-spanner/blob/72efac10bcab961732cca234312908fa1b5bc3dd/src/batch-transaction.js#L116) to include `region_tag`, But I see Document generation in `nodejs-spanner` repository as well

**Comment by sduskis** [comment]
@jmdobry, I created a new branch on `nodejs-bigtable` called `docs-test`.  Can you please try it out?

**Comment by vijay-qlogic** [comment]
Hi Jason,

Can you please verify the "example" section while generating the API Docs.
Because in my case "region_tag" do not pulls the example code.

Sharing the screen-shot here.

**Comment by sduskis** [comment]
@vijay-qlogic, this was verified.

**Pull request merged**